x-pack/plugins/security_solution/public/common/components/event_details/cti_details/helpers.tsx

```diff
@@ -0,0 +1,127 @@
+  data: TimelineEventsDetailsItem[]
+): TimelineEventsDetailsItem[][] => {
+  const threatIndicatorField = data.find(
+    ({ field, originalValue }) => field === INDICATOR_DESTINATION_PATH && originalValue
+  );
+  if (!threatIndicatorField) {
+    return [];
+  }
+
+  const { originalValue } = threatIndicatorField;
+  const enrichmentStrings = Array.isArray(originalValue) ? originalValue : [originalValue];
+
+  return enrichmentStrings
+    .map((enrichmentString) => {
+      try {
+        const enrichment = JSON.parse(enrichmentString);
+        return getDataFromSourceHits(enrichment);
+      } catch (e) {
+        return null;
+      }
+    })
+    .filter(isEventDetailsItem);
+};
+
+export const timelineDataToEnrichment = (data: TimelineEventsDetailsItem[]): CtiEnrichment =>
+  data.reduce<CtiEnrichment>((acc, item) => {
+    acc[item.field] = item.originalValue;
+    return acc;
+  }, {});
+
+export const getTooltipTitle = (type: string | undefined) =>
+  isInvestigationTimeEnrichment(type)
+    ? i18n.INVESTIGATION_TOOLTIP_TITLE
+    : i18n.INDICATOR_TOOLTIP_TITLE;
+
+export const getTooltipContent = (type: string | undefined) =>
+  isInvestigationTimeEnrichment(type)
+    ? i18n.INVESTIGATION_TOOLTIP_CONTENT
+    : i18n.INDICATOR_TOOLTIP_CONTENT;
+
+export const getFirstElement: <T = unknown>(array: T[] | undefined) => T | undefined = (array) =>
+  array ? array[0] : undefined;
+
+export const getEnrichmentValue = (enrichment: CtiEnrichment, field: string) =>
+  getFirstElement(enrichment[field]) as string | undefined;
+
+/**
+ * These fields (e.g. 'x') may be in one of two keys depending on whether it's
+ * a new enrichment ('threatintel.indicator.x') or an old indicator alert
+ * (simply 'x'). Once enrichment has been normalized and we support the new ECS
+ * fields, this value should always be 'indicator.x';
+ */
+export const getShimmedIndicatorValue = (enrichment: CtiEnrichment, field: string) =>
+  getEnrichmentValue(enrichment, field) ||
+  getEnrichmentValue(enrichment, `${DEFAULT_INDICATOR_SOURCE_PATH}.${field}`);
+
+const buildEnrichmentId = (enrichment: CtiEnrichment): string => {
+  const matchedId = getEnrichmentValue(enrichment, MATCHED_ID);
+  const matchedField = getEnrichmentValue(enrichment, MATCHED_FIELD);
+  return `${matchedId}${matchedField}`;
+};
+/**
+ * This function receives an array of enrichments and removes
+ * investigation-time enrichments if that exact indicator already exists
+ * elsewhere in the list.
+ *
+ * @param enrichments {@type CtiEnrichment[]}
+ */
+export const filterDuplicateEnrichments = (enrichments: CtiEnrichment[]): CtiEnrichment[] => {
+  if (enrichments.length < 2) {
+    return enrichments;
+  }
+  const enrichmentsById = groupBy(enrichments, buildEnrichmentId);
+
+  return Object.values(enrichmentsById).map(
+    (enrichmentGroup) =>
+      enrichmentGroup.find(
+        (enrichment) => !isInvestigationTimeEnrichment(getEnrichmentValue(enrichment, MATCHED_TYPE))
```

Comment:
Correct, this is a "hack" to deal with the existing indicator alerts: since we only deal with these two types of enrichment, I'm grouping them into two buckets:

* `isInvestigationTimeEnrichment == true`
    * this will only be true for the enrichments returned from our new endpoint
* `isInvestigationTimeEnrichment == false`
    * this covers all other alerts, which for now we're assuming came from the indicator match rule